reference-implementation/lib/transform-stream.js

```diff
@@ -262,12 +238,32 @@ class TransformStreamSource {
 
     transformStream._readableController = c;
 
-    return this._startPromise;
+    // delay the first source.pull call until there is backpressure
+    return this._startPromise.then(() => {
```

Comment:
I was unclear. I wanted to say the following.

You can pass a promise already then-ed with this callback in TransformStream's constructor to the constructor of TransformStreamSource, and then you just return the passed promise stored in _startPromise (which is already then-ed).